server/src/main.rs

```diff
@@ -1,61 +1,69 @@
 #[macro_use]
 extern crate diesel;
+#[macro_use]
+extern crate log;
 
 pub mod model;
 pub mod schema;
 
-use std::sync::Arc;
-use warp::Filter;
-use dotenv::dotenv;
+use diesel::r2d2;
 use diesel::PgConnection;
+use dotenv::dotenv;
 use std::env;
-use diesel::connection::Connection;
-use tokio::sync::Mutex;
+use warp::Filter;
 
-pub type Db = Arc<Mutex<PgConnection>>;
+type Pool = r2d2::Pool<r2d2::ConnectionManager<PgConnection>>;
 
-pub async fn establish_connection() -> Db {
+fn get_connection_pool() -> Pool {
     dotenv().ok();
-
-    let database_url = env::var("DATABASE_URL")
-        .expect("DATABASE_URL must be set");
-    let pgc = PgConnection::establish(&database_url)
-        .expect(&format!("Error connecting to {}", database_url));
-    Arc::new(Mutex::new(pgc))
+    let database_url = env::var("DATABASE_URL").expect("DATABASE_URL must be set");
```

Comment:
I don't want to use `.expect` really. Do you think it'd be a lot of work to introduce a service error?

Reply:
I mean this should just prevent the service from starting no?